src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -2,18 +2,135 @@
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
 
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private int threshold;
+    private Node<K, V>[] table;
+    private int size;
+
     @Override
     public void put(K key, V value) {
-
+        int hashCodeNode = hashCode(key);
+        Node<K, V> lastElement;
+        if (table == null) {
+            table = (Node<K, V>[]) new Node[1 << 4];
+            threshold = (int) (table.length * DEFAULT_LOAD_FACTOR);
+            table[getTableIndex(key, table.length)] = new Node<>(hashCodeNode, key, value, null);
+            size = 1;
+            return;
+        }
+        int tableIndex = getTableIndex(key, table.length);
+        Node<K, V> element = table[tableIndex];
+        if (element == null) {
+            table[tableIndex] = new Node<>(hashCodeNode, key, value, null);
+        } else {
+            do {
+                if (key == null && element.key == null
+                        || element.key != null && element.key.equals(key)) {
+                    element.value = value;
+                    return;
+                }
+                lastElement = element;
+                element = element.next;
+            } while (element != null);
+            lastElement.next = new Node<>(hashCodeNode, key, value, null);
```

Comment:
```suggestion
            while (element != null) {
                if (Objects.equals(element.key, key)) {
                    element.value = value;
                    break;
                }
                if (element.next == null) {
                    element.next = new Node<>(hashCodeNode, key, value, null);
                    size++;
                    break;
                }
                element = element.next;
            }
```
lets simplify a little bit, this will prevent from using additional variable